daemon/src/model.rs

```diff
@@ -692,17 +690,145 @@ impl str::FromStr for OpeningFee {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         let amount_sat: u64 = s.parse()?;
-        Ok(OpeningFee { amount_sat })
+        Ok(OpeningFee {
+            fee: Amount::from_sat(amount_sat),
+        })
     }
 }
 
+impl_sqlx_type_display_from_str!(OpeningFee);
+
 impl Default for OpeningFee {
     fn default() -> Self {
-        Self::new(Amount::ZERO)
+        Self { fee: Amount::ZERO }
     }
 }
 
-impl_sqlx_type_display_from_str!(OpeningFee);
+impl From<OpeningFee> for Amount {
+    fn from(opening_fee: OpeningFee) -> Self {
+        opening_fee.fee
+    }
+}
+
+/// Funding fee as defined by position and rate
+///
+/// Position and rate define if the funding rate is to be added or subtracted when summing up the
+/// fees.
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct FundingFee {
+    #[serde(with = "bdk::bitcoin::util::amount::serde::as_sat")]
+    fee: Amount,
+    rate: FundingRate,
```

Comment:
🔧 I would prefer it if we just stored the sign (using an `enum` or even a `bool`) instead of the funding rate. Otherwise we're storing more information than what we need and this data type is a bit confusing (i.e. I needed to verify that the `rate` field is only used for the sign). Perhaps we can use the `FundingRate` type only in the constructor?

Reply:
@da-kami @klochowicz Thoughts?

I could be convinced that a bool or enum is better.

Looking at https://doc.rust-lang.org/std/primitive.i64.html#method.is_positive and https://doc.rust-lang.org/std/primitive.i64.html#method.is_negative, using the sign might be problematic because it returns `false` in both cases for `0`, meaning we should actually forbid constructing a `FundingFee` with a `FundingRate` of `0` or at least special case this in the constructor to handle it correctly.

Can we check with a test on what happens if we use a `FundingRate` of `0`?

Regarding your concern @luckysori, one we figured it might be useful for the future to record the actual funding rates that were used to calculate the fees.

Reply:
> Regarding your concern [...] we figured it might be useful for the future to record the actual funding rates that were used to calculate the fees.

That sounds fair enough. With that in mind I am not against keeping the field, even though we're not using it that way yet.

Reply:
The `FundingRate` is directly related to the `FundingFee`. I think we should store this information together, even if we don't "use" it at the moment.